src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -30146,42 +30151,53 @@ public void SuppressNullableWarning_TypeParameters_01()
             var source =
 @"class C
 {
-    static void F1<T1>()
+    static void F1<T1>(T1 t1)
     {
-        default(T1).ToString();
+        default(T1).ToString(); // 1
         default(T1)!.ToString();
+        t1.ToString(); // 2
+        t1!.ToString();
     }
-    static void F2<T2>() where T2 : class
+    static void F2<T2>(T2 t2) where T2 : class
     {
-        default(T2).ToString();
+        default(T2).ToString(); // 3
         default(T2)!.ToString();
+        t2.ToString();
+        t2!.ToString();
     }
-    static void F3<T3>() where T3 : struct
+    static void F3<T3>(T3 t3) where T3 : struct
     {
         default(T3).ToString();
-        default(T3)!.ToString();
+        default(T3)!.ToString(); // 4
+        t3.ToString();
+        t3!.ToString(); // 5
     }
 }";
             var comp = CreateCompilation(
                 new[] { source, NonNullTypesTrue, NonNullTypesAttributesDefinition },
                 parseOptions: TestOptions.Regular8);
             comp.VerifyDiagnostics(
                 // (5,9): warning CS8602: Possible dereference of a null reference.
-                //         default(T1).ToString();
+                //         default(T1).ToString(); // 1
                 Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "default(T1)").WithLocation(5, 9),
-                // (10,9): warning CS1720: Expression will always cause a System.NullReferenceException because the default value of 'T2' is null
-                //         default(T2).ToString();
-                Diagnostic(ErrorCode.WRN_DotOnDefault, "default(T2).ToString").WithArguments("T2").WithLocation(10, 9),
-                // (10,9): warning CS8602: Possible dereference of a null reference.
-                //         default(T2).ToString();
-                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "default(T2)").WithLocation(10, 9),
-                // (16,9): error CS8624: The ! operator can only be applied to reference types.
-                //         default(T3)!.ToString();
-                Diagnostic(ErrorCode.ERR_NotNullableOperatorNotReferenceType, "default(T3)!").WithLocation(16, 9));
+                // (7,9): warning CS8602: Possible dereference of a null reference.
+                //         t1.ToString(); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "t1").WithLocation(7, 9),
+                // (12,9): warning CS1720: Expression will always cause a System.NullReferenceException because the default value of 'T2' is null
+                //         default(T2).ToString(); // 3
+                Diagnostic(ErrorCode.WRN_DotOnDefault, "default(T2).ToString").WithArguments("T2").WithLocation(12, 9),
+                // (12,9): warning CS8602: Possible dereference of a null reference.
+                //         default(T2).ToString(); // 3
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "default(T2)").WithLocation(12, 9),
```

Comment:
>WRN_NullReferenceReceiver [](start = 37, length = 25)

Seems strange that we produce this almost-duplicate warning on `default(T2)`. The warning above feels stronger, so this one seems unnecessary.